src/puppeteer_pool.js

```diff
@@ -303,4 +303,38 @@ export default class PuppeteerPool {
             .all(closePromises)
             .catch(err => log.exception(err, 'PuppeteerPool: cannot close the browsers'));
     }
+
+    /**
+     * Finds a PuppeteerInstance given a Puppeteer Browser running in the instance.
+     * @param {Puppeteer.Browser} browser
+     * @ignore
+     */
+    _findInstanceByBrowser(browser) {
```

Comment:
I'd just split that line 314 to dont have to array functions inside. Otherwise looks good.

```
const activeInstances = Object.values(this.activeInstances);
const browserPromises = activeInstances.map(instance => instance.browserPromise);

return Promise.filter(browserPromises, savedBrowser => browser === savedBrowser );
```

Reply:
I implemented it, debugged it for 30 mins and then realized it can't work because we need to filter the instances based on their internal browsers. If I filter the browsers, I will not get a `PuppeteerInstance` back, but a `Browser`. So I suggest keeping the previous implementation :)